pkg/tnf/interactive/pty.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package interactive
+
+import (
+	"bytes"
+	"html/template"
+	"io/ioutil"
+	"time"
+
+	expect "github.com/ryandgoulding/goexpect"
+	"github.com/test-network-function/test-network-function/pkg/jsonschema"
+	"github.com/xeipuuv/gojsonschema"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// PTYSchemaFileName is the filename for the generic pty JSON schema.
+	PTYSchemaFileName = "generic-pty.schema.json"
+)
+
+// SpawnGenericPTYFromYAMLFile attempts to spawn an interactive PTY given the supplied file based on
+// generic-pty.schema.json.  If the supplied file does not conform to the schema, creation fails and the result is
```

Comment:
It actually is.  The `schemaPath` is propagated because it differs based on program entrypoint (i.e., from Ginkgo, from Unit Tests, From the `jsontest-cli` executable).  This is an unfortunate artifact of utilizing a compiled language.  I will clarify that the file should always be the relative path to `generic-pty.schema.json`.